tools/test_correctness.cpp

```diff
@@ -0,0 +1,90 @@
+#include <fstream>
+#include <unordered_map>
+#include <iostream>
+#include <thread>
+#include <algorithm>
+#include <chrono>
+
+#include <cc_sketch_alg.h>
+#include <mat_graph_verifier.h>
+
+struct CorrectnessResults {
+  size_t num_failures = 0;
+  std::vector<size_t> num_round_hist;
+};
+
+CorrectnessResults test_path_correctness(size_t num_vertices, size_t num_graphs,
+                                         size_t samples_per_graph) {
+  CorrectnessResults results;
+
+  size_t num_rounds = Sketch::calc_cc_samples(num_vertices);
+  for (size_t r = 0; r < num_rounds; r++)
+    results.num_round_hist.push_back(0);
+
+  std::vector<node_id_t> vertices;
+  for (size_t i = 0; i < num_vertices; i++) {
+    vertices.push_back(i);
+  }
+
+  size_t t = std::chrono::steady_clock::now().time_since_epoch().count();
+  std::mt19937_64 gen(t);
+
+  for (size_t g = 0; g < num_graphs; g++) {
+    size_t edge_seed = gen();
```

Comment:
there's no parallelism here-- can we not use gen for all of our randomness? Why do we reseed with a random number every iteration?

Randomly generating seeds (as well as sequential seeds) are both a bit messy. (birthday paradox and risk of correlation)

I don't think it matters here too much, but every now and then I enter into the raddit hole of using RNG to seed RNG and exit without a clear indication of what's best. This post claims to know more than I do.

https://stats.stackexchange.com/questions/233061/best-way-to-seed-n-independent-random-number-generators-from-1-value